### ISSUE TITLE
EncodedValueRewriter's array rewriter

### ISSUE THREAD

**Issue by Lanchon** [JesusFreke/smali#741]
though not a bug, i think this line:

https://github.com/JesusFreke/smali/blob/bf65e575a0fd7f03c41aa56d7a49bf0cda1da06a/dexlib2/src/main/java/org/jf/dexlib2/rewriter/EncodedValueRewriter.java#L127

would be 'more correct' like this:
```
return RewriterUtils.rewriteList(rewriters.getEncodedValueRewriter(), arrayEncodedValue.getValue());
```

the expectation is that rewriters delegate to the module, not to themselves just because they happen to match the needed type of rewriter.

**Comment by Lanchon** [comment]
to give you a real-world example of this subtle difference, [this code](https://github.com/DexPatcher/dexpatcher-tool/blob/ccb73c5e09f044e74b3afa7ac145bdb3a8207dd7/tool/src/main/java/lanchon/dexpatcher/transform/decoder/DexDecoderModule.java#L204-L219) -as i wrote it- was expected to rewrite an annotation element value only if the element value is a string. however the code unexpectedly rewrites arrays of [arrays of...] strings too.

**Comment by JesusFreke** [comment]
Fixed by cb37770c1cfec0c72336b0c335d82365852d8be3
